RecoHI/HiJetAlgos/plugins/SameEventVoronoiAlgorithm.cc

```diff
@@ -0,0 +1,399 @@
+	const bool exclusion_by_constituent)
+{
+	std::vector<fastjet::PseudoJet> pseudojet;
+
+	for (std::vector<particle_t>::const_iterator iterator =
+			 _event.begin();
+		 iterator != _event.end(); iterator++) {
+		pseudojet.push_back(fastjet::PseudoJet(
+			iterator->momentum.px(),
+			iterator->momentum.py(),
+			iterator->momentum.pz(),
+			iterator->momentum.energy()));
+		pseudojet.back().set_user_index(
+			iterator - _event.begin());
+	}
+
+	fastjet::JetDefinition
+		jet_definition(fastjet::antikt_algorithm,
+					   antikt_distance);
+	fastjet::ClusterSequence
+		cluster_sequence(pseudojet, jet_definition);
+	std::vector<fastjet::PseudoJet> jet =
+		cluster_sequence.inclusive_jets();
+
+	exclusion_density = std::vector<bool>(_event.size(), false);
+	exclusion_flow = std::vector<bool>(_event.size(), false);
+
+	for (std::vector<fastjet::PseudoJet>::const_iterator
+			 iterator_jet = jet.begin();
+		 iterator_jet != jet.end(); iterator_jet++) {
+		std::vector<fastjet::PseudoJet> constituent =
+			cluster_sequence.constituents(*iterator_jet);
+		double perp_resummed = 0;
+
+		for (std::vector<fastjet::PseudoJet>::const_iterator
+				 iterator_constituent = constituent.begin();
+			 iterator_constituent != constituent.end();
+			 iterator_constituent++) {
+			perp_resummed +=
+				_event[iterator_constituent->user_index()].
+				momentum_perp_subtracted_unequalized;
+		}
+
+		bool jet_excluded = perp_resummed >= exclusion_perp_min;
+
+		if (fake_reject) {
+			// ATLAS E_T^max/<E_T> fake rejection using 0.1x0.1
+			// pseudotower, as in Phys. Lett. B 719 (2013) 222 (left
+			// column).
+
+			std::map<std::pair<int, int>, double> pseudotower;
+
+			for (std::vector<fastjet::PseudoJet>::const_iterator
+				 iterator_constituent = constituent.begin();
+			 iterator_constituent != constituent.end();
+			 iterator_constituent++) {
+				const int int_pseudorapidity_positive =
+					eta2ieta(iterator_constituent->pseudorapidity());
+				if (int_pseudorapidity_positive >= 0) {
+					const int int_pseudorapidity =
+						(iterator_constituent->pseudorapidity() < 0 ? -1 : 1) *
+						int_pseudorapidity_positive;
+					const int int_azimuth =
+						phi2iphi(iterator_constituent->phi_std(),
+								 int_pseudorapidity);
+
+					pseudotower[
+						std::pair<int, int>(int_pseudorapidity, int_azimuth)] +=
+						_event[iterator_constituent->user_index()].
+						momentum_perp_subtracted_unequalized;
+				}
+			}
+
+			double et_max = 0;
+			double et_mean = 0;
+
+			for (std::map<std::pair<int, int>, double>::const_iterator
+					 iterator_pseudotower = pseudotower.begin();
+				 iterator_pseudotower != pseudotower.end();
+				 iterator_pseudotower++) {
+				et_max = std::max(et_max, iterator_pseudotower->second);
+				et_mean += iterator_pseudotower->second;
+			}
+			et_mean /= pseudotower.size();
+
+			jet_excluded &= (et_max >= _same_event_fake_reject_et_max &&
+							 et_max / et_mean >=
+							 _same_event_fake_reject_et_max_over_mean);
```

Comment:
More protection against division by 0 needed. Please check pseudotower.size() > 0 and et_mean > 0.